### PR TITLE
ATB-1161 | Establish secure pipelines for core, main, and alarm template in Integration and Production env

### PR DESCRIPTION
**Pull request by MinaMahdi**
## Proposed changes

ATB-1161: Establish secure pipelines for core, main, and alarm template in Integration and Production env

### What changed

Establishing `ais-core-pipeline`, `ais-main-pipeline`, and `ais-alarm-pipeline` for integration and production envs. 
New Parameter added - `ArtifactSourceBucketEventTriggerRoleArn` that is now required - https://govukverify.atlassian.net/wiki/spaces/PLAT/pages/3626042836/How+to+migrate+pipeline+to+an+event-based+mechanism+for+change+detection 
 
### Why did it change

To set up our secure pipelines  `ais-core-pipeline`, `ais-main-pipeline`, and `ais-alarm-pipeline` for integration and production envs. 
To allow build to promote to staging and staging to promote to integration, but promotion for production is switched off. 

### Issue tracking

- [ATB-1161](https://govukverify.atlassian.net/browse/ATB-1161)

## Testing

Integration 
Screenshot 2023-11-13 at 14 30 15

Production 
Screenshot 2023-11-13 at 14 30 49
When clicking on Failed Current execution  > 
Screenshot 2023-11-13 at 14 51 17 

## Checklists
- [ ] Did not commit any not-required changes to the `src/infra/**/samconfig.toml`
- [ ] Tested changes and included test evidence in the PR, if appropriate
- [ ] Included all required tags and other properties for any new resources in the SAM template
- [ ] Ensured that any new resources in the SAM Template follow appropriate naming conventions
- [ ] Ensured that naming of new resources is compatible with deploying multiple stacks with custom stack names during development
- [ ] Ensured that no log lines include PII or other sensitive data
- [ ] Implemented unit testing for any new logic implemented, if appropriate
- [ ] Ensured that all commits in this PR are signed
- [ ] Ensured appropriate code coverage is maintained by unit tests

[ATB-1161]: https://govukverify.atlassian.net/browse/ATB-1161?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by TaherKapasi — APPROVED** [review]

File: stack-orchestration/configuration/di-account-interventions-integration-admin/ais-alarm-pipeline/parameters.json [hunk 1]
```diff
@@ -0,0 +1,90 @@
+[
+  {
+    "ParameterKey": "AllowedAccounts",
+    "ParameterValue": "324281879537"
```
Comment (domantas-gds) [inline stack-orchestration/configuration/di-account-interventions-integration-admin/ais-alarm-pipeline/parameters.json:4]: I don't think production should be added here. It's staging to integration and production. But we do not want to promote to production yet.

There it's just in staging pipeline allow integration to read from the promotion bucket.

Comment (domantas-gds) [reply stack-orchestration/configuration/di-account-interventions-integration-admin/ais-alarm-pipeline/parameters.json:4]: And same will apply for other integration pipelines :)

File: stack-orchestration/configuration/di-account-interventions-integration-admin/ais-alarm-pipeline/parameters.json [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  },
+  {
+    "ParameterKey": "ArtifactSourceBucketArn",
+    "ParameterValue": "arn:aws:s3:::ais-alarm-pipeline-artifactpromotionbucket-9zjxdwt6itx4"
```
Comment (domantas-gds) [inline stack-orchestration/configuration/di-account-interventions-integration-admin/ais-alarm-pipeline/parameters.json:8]: What promotion bucket is this? Should be that of the AIS alarm pipeline in Staging. But in order for that bucket to be created you must enable promotion on each staging pipeline.

File: stack-orchestration/configuration/di-account-interventions-integration-admin/ais-alarm-pipeline/parameters.json (hunk 1 above)
Comment (MinaMahdi) [reply stack-orchestration/configuration/di-account-interventions-integration-admin/ais-alarm-pipeline/parameters.json:4]: Awesome, thanks Dom will remove 🙌

**Review by domantas-gds — APPROVED** [review]

**Pull request merged**